packages/database/realtime/database.service.ts

```diff
@@ -0,0 +1,312 @@
+import {Injectable} from "@nestjs/common";
+import {
+  ChangeStream,
+  DatabaseService,
+  Document,
+  FilterQuery,
+  ObjectId
+} from "@spica-server/database";
+import {Observable, Subject, Subscription} from "rxjs";
+import {bufferTime, filter, share} from "rxjs/operators";
+import {levenshtein} from "./levenshtein";
+import {late} from "./operators";
+import {ChunkKind, StreamChunk} from "./stream";
+
+/**
+ * Converts the filter to not form
+ * @example { 'subfilter': true } -> { 'fullDocument.subfilter': { '$ne': true } }
+ * @example { 'fullDocument.stars': { '$gt': 1 } }  -> { 'fullDocument.stars': { '$not': { $gt: 1 } } }
+ */
+function reverseFilter<T>(filter: FilterQuery<T>) {
+  return Object.keys(filter).map(key => {
+    const value = filter[key];
+    return {
+      [`fullDocument.${key}`]: typeof value == "object" ? {$not: value} : {$ne: value}
+    };
+  });
+}
+
+function getSortedKeys(sort: {[k: string]: -1 | 1}) {
```

Comment:
@thesayyn Can you please explain how keys are sorted with this function?

Reply:
As a matter of fact, this function just returns the sorted keys of the cursor. 
Shall I rename it to `getSortedKeysOfTheCursor`? Would it make it more understandable?

Reply:
@thesayyn I don't get what "sorted" means. In what manner do we sort the keys and why do we sort them? :)

Reply:
In order to sort the cursor, the client provides the name(s) of the properties that he/she want to sort. 

A complete sorting condition looks like this
`{test:-1, anotherprop:1}`.

So this method returns only the keys of the sorting as an array.